### DISCUSSION TITLE
Shift camera position with followUserLocation

### DISCUSSION THREAD

**Discussion by trevorfrost** [rnmapbox/maps#1622] [Q&A]
Is it possible to shift the camera so that user's location appears at the bottom of the screen?

How currently is:
<img src="https://user-images.githubusercontent.com/75694984/140621720-4fafe693-2f02-4fa9-a73a-c869c19ee9f7.png" alt="alt text" width="190" height="300">

What I'm trying to achieve:
<img src="https://user-images.githubusercontent.com/75694984/140621742-27d9e287-1bf3-491e-abec-1abc9a4a19ac.png" alt="alt text" width="190" height="300">

**Comment by ferdicus** [comment]
You can check our the `example` project. 
It has a scene called `UserLocation` => `SetUserLocationVerticalAlignment` which uses `contentInset` to achieve this:
https://github.com/react-native-mapbox-gl/maps/blob/4060e7848c363bba46398dc1bcdebc3f5838bb87/example/src/examples/UserLocation/SetUserLocationVerticalAlignment.js#L52

**Reply by trevorfrost** [reply]
Yep this works, you must use state, hard-coded values don't work.

There is another issue toggling when `[followUserLocation]` with `followPitch` or `followZoomLevel` set, after toggling once breaks and doesn't continue to follow, should I open an issue for this?

**Reply by trevorfrost** [reply]
Demo video:


https://user-images.githubusercontent.com/75694984/141023616-a9948a03-ccfc-4d3b-8789-fa32212b00b0.mp4

**Reply by trevorfrost** [reply]
Current code:

```js
<MapboxGL.Camera
  followPitch={50}
  followUserLocation={followUser}
  followUserMode={MapboxGL.UserTrackingModes.FollowWithHeading}
  followZoomLevel={17}
/>

// Toggle
<Button
  title={followUser ? 'Un-Follow' : 'Follow'}
  onPress={() => {
    setFollowUser(f => !f);
  }}
/>
```

**Reply by trevorfrost** [reply]
Working demo, `no followPitch` and `no followZoomLevel`:

https://user-images.githubusercontent.com/75694984/141025094-00abea9d-932b-4837-9769-e24a9986c4c8.mp4

**Reply by ferdicus** [reply]
I think we have some open issues for that in the repo already, can you review the open tickets and see if you find one that matches. 
If not, feel free to open a new ticket. 

Cheers 🍻

**Reply by trevorfrost** [reply]
@ferdicus I didn't find any open issue that match mine, but as a workaround for now, is there a way to destroy `MapboxGL.Camera` completely and re-initialize it?

I feel like that would be a basic workaround for now until it's sorted.

Also for some reason `MapboxGL.UserLocation#onUpdate` subscription doesn't seem to cancel on hot reload, so it just keeps adding more and more subscribers.
